connector-runtime/spring-boot-starter-camunda-connectors/src/test/java/io/camunda/connector/runtime/app/EnvVarContextInitializer.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.connector.runtime.app;
+
+import java.lang.reflect.InvocationTargetException;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.env.EnvironmentPostProcessor;
+import org.springframework.core.env.ConfigurableEnvironment;
+
+/**
+ * This is a hack to inject the "environment variables" before the Configurations are loaded, so
+ * that the env vars are considered during construction of the inbound and outbound factories.
+ */
```

Comment:
I dont fully understand this comment: 
> hat the env vars are considered during construction of the inbound and outbound factories.

Cant we model the "filters" by having dedicated beans that provide this kind of functionality so they can easily be mocked in the tests?

Reply:
That's a very elegant idea. This would allow for a more powerful features down the line